src/systems/environment_preload/EnvironmentPreload.cc

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "EnvironmentPreload.hh"
+
+#include <array>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gz/common/CSVStreams.hh>
+#include <gz/common/DataFrame.hh>
+#include <gz/common/Filesystem.hh>
+
+#include <gz/plugin/Register.hh>
+
+#include "gz/sim/components/Environment.hh"
+#include "gz/sim/components/World.hh"
+#include "gz/sim/Util.hh"
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+//////////////////////////////////////////////////
+class gz::sim::systems::EnvironmentPreloadPrivate
+{
+  public: bool loaded{false};
+
+  public: std::shared_ptr<const sdf::Element> sdf;
+};
+
+//////////////////////////////////////////////////
+EnvironmentPreload::EnvironmentPreload()
+  : System(), dataPtr(new EnvironmentPreloadPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+EnvironmentPreload::~EnvironmentPreload() = default;
+
+//////////////////////////////////////////////////
+void EnvironmentPreload::Configure(
+    const Entity &/*_entity*/,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &/*_ecm*/,
+    EventManager &/*_eventMgr*/)
+{
+  this->dataPtr->sdf = _sdf;
+}
+
+//////////////////////////////////////////////////
+void EnvironmentPreload::PreUpdate(
+  const gz::sim::UpdateInfo &,
+  gz::sim::EntityComponentManager &_ecm)
+{
+  if (!std::exchange(this->dataPtr->loaded, true))
```

Comment:
Is there a reason we implement `PreUpdate` when we probably could get away with doing this in `Configure`?

Reply:
@arjo129 it used to be that way. Thing is, we can't query the world SDF (root) path until it's fully loaded, and that has to wait for all world plugins (including this one) to load and configure.